modules/repository/repo.go

```diff
@@ -127,11 +127,24 @@ func MigrateRepositoryGitData(ctx context.Context, u *models.User, repo *models.
 	}
 
 	if opts.Mirror {
+		var interval = setting.Mirror.DefaultInterval
+		if opts.MirrorInterval != "" {
+			interval, err = time.ParseDuration(opts.MirrorInterval)
+			if err != nil {
+				log.Error("Failed to set Interval: %v", err)
+				return repo, err
+			}
+			if interval != 0 && interval < setting.Mirror.MinInterval {
+				err := fmt.Errorf("Interval is set below Minimum Interval")
+				log.Error("Interval is too frequent")
+				return repo, err
+			}
+		}
 		if err = models.InsertMirror(&models.Mirror{
 			RepoID:         repo.ID,
-			Interval:       setting.Mirror.DefaultInterval,
+			Interval:       interval,
 			EnablePrune:    true,
-			NextUpdateUnix: timeutil.TimeStampNow().AddDuration(setting.Mirror.DefaultInterval),
+			NextUpdateUnix: timeutil.TimeStampNow().AddDuration(interval),
```

Comment:
Need to include some way of expressing no update.

Reply:
Yep, when you pass in zero this has been resolved to say don't do an update properly.

Reply:
thanks